src/python/gudhi/simplex_tree.pyx

```diff
@@ -612,7 +612,7 @@ cdef class SimplexTree:
         """
         self.get_ptr().expansion_with_blockers_callback(max_dim, callback, <void*>blocker_func)
 
-    def persistence(self, homology_coeff_field=11, min_persistence=0, persistence_dim_max = False):
+    def persistence(self, homology_coeff_field=11, min_persistence=0, persistence_dim_max = False, output_type = 'old'):
```

Comment:
-- Not completely related to this PR, but I think we should do type annotations here, i.e., 
```python
from typing import Optional,Literal, Union
def persistence(self, homology_coeff_field:int=11, min_persistence:float=0., persistence_dim_max:Optional[int] = None, output_type:Literal['old','array by dimension'] = 'old') -> list:
    pass
```
especially to have auto-completion for long arguments such as "array by dimension". 
I also changed the `False` to a `None`.